packages/core/src/pipeline/config/validation/imageProviderBeforeType.validator.ts

```diff
@@ -0,0 +1,30 @@
+import { IStageOrTriggerValidator, IValidatorConfig, PipelineConfigValidator } from './PipelineConfigValidator';
+import { ICluster, IPipeline, IStage, IStageOrTriggerTypeConfig } from '../../../domain';
+
+export interface IImageProviderBeforeTypeValidationConfig extends IValidatorConfig {
+  triggerTypes?: string[];
+  message: string;
+}
+
+export class ImageProviderBeforeTypeValidator implements IStageOrTriggerValidator {
+  public validate( 
+    pipeline: IPipeline,
+    stage: IStage,
+    validator: IImageProviderBeforeTypeValidationConfig,
+    _config: IStageOrTriggerTypeConfig,
+  ): string {
+
+    const triggersToTest = pipeline.triggers || [];
+    const hasImageProvider = Boolean(triggersToTest.length) && triggersToTest.every((trigger) => (validator.triggerTypes || []).includes(trigger.type));
+
+    const hasCustomImage = (stage.clusters || []).every((cluster: ICluster) => cluster.imageId && cluster.imageId !== '${trigger.properties.imageName}');
```

Comment:
1. `cluster` is not a property of `stage`, but `Istage` also has `[k: string]: any;` in its definition. I'm just wondering why :)
2. You can do `stage.clusters?.every...` instead

Reply:
1. This is because all the stage types have such different metadata. A deploy stage has `cluster`, which falls under `[k: string]: any`, but most other stages do not. The `IStage` types could definitely use an overhaul for some more explicit typing (i.e. `IDeployStage extends IStage`). 
2. 👍🏼